### ISSUE TITLE
Not loading EPG info after two days

### ISSUE THREAD

**Issue by Sjasie** [yohaybn/HomeAssistant-EPG#5]
Hi there, the json attributes delete when time goes and don't update unless I restart HA. Cold you make a service call to pull new files?

**Comment by yohaybn** [comment]
Hi,
I didn't understand what happened. can you elaborate? 
which file you use and which channel?
anyway you can use home assistant update_entity service.

**Comment by Sjasie** [comment]
Hi there,

I tried using the update entity but it doesnt load data. When I restart Hass it does load new data.

It only loads xml data untill next day.

**Comment by Keysersose12** [comment]
I'm seeing this too this week. For example it downloaded in on Tuesday evening and by Thursday night it still had not downloaded new epg. I restarted that night and it downloaded again but by Saturday morning it was back to no new epg data. Is there any way to reload the integration without restarting home assistant? I have manually downloaded epg but it wont read the new epg file until i restart upon which it downloads the xml file itself again.

**Comment by yohaybn** [comment]
hi.
I updated the integration.
please update and check if it works for.
thank you

**Comment by Keysersose12** [comment]
Yes looks to be working fine with 2.2, thanks for adding the download service. The only issue i came across and maybe you have it corrected in the latest updates or maybe due to my setup was when installing the integration and picking the channels, the popup window was small so it was a bit slow to scroll through the channels when using the full channel list but just selecting which were needed. I havent got around to using the custom epg option yet. Anyways cheers the for the updates.

**Comment by yohaybn** [comment]
Hi.
Its work fine on my development environment and my android app but I have the same issue on my production machine.
I hope to fix it someday :)
